src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -267,8 +267,13 @@ private fun inferArrayType(expr: RsArrayExpr): Ty {
     } else {
         val elements = expr.arrayElements
         if (elements.isNullOrEmpty()) return TySlice(TyUnknown)
+
+        var elementType: Ty = TyUnknown;
         // '!!' is safe here because we've just checked that elements isn't null
-        elements!![0].type to elements.size
+        for (e in elements!!) {
```

Comment:
An interesting question. I think it's OK to inspect all the elements of the array. Of course there are some awful code-generated cases with huge arrays which can hit us hard, but we probably will calculate the types of elements anyway for highlighting.